test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs

```diff
@@ -2256,6 +1629,24 @@ public void Dispose()
             }
         }
 
+        private static async Task<X509Certificate2> GetExpiringCertificateAsync(SigningTestFixture fixture)
+        {
+            CertificateAuthority ca = await fixture.GetDefaultTrustedCertificateAuthorityAsync();
+
+            AsymmetricCipherKeyPair keyPair = SigningTestUtility.GenerateKeyPair(publicKeyLength: 2048);
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+            var issueOptions = new IssueCertificateOptions()
+            {
+                KeyPair = keyPair,
+                NotAfter = now.AddSeconds(10),
```

Comment:
In some cases (like expiration and revocation), some time padding is necessary to enable the test to generate a signature while the certificate is still valid.  NuGet will not generate a signature if a certificate is expired or revoked.  Because these things take time and throughput of the test machine is subject to current load, the padding is on the safe side.  If we reduce the buffer too much, the test will randomly fail and be flakey.

Please open a suggestion in the engineering repo with your ideas on streamlining this.